From e801ccb2633e3f5842ed6d160af0e6e5cce44c77 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 2 Sep 2014 22:38:55 +0200
Subject: [PATCH 1/4] Fix warnings related to subdir-objects

Use AM_INIT_AUTOMAKE in configure.ac to define options such as
subdir-objects, instead of AUTOMAKE_OPTIONS in Makefile.am, as it
allows to avoid warnings during autoreconf:

examples/Makefile.am:25: warning: source file '../lib/connect.c' is in a subdirectory,
examples/Makefile.am:25: but option 'subdir-objects' is disabled

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 Makefile.am  | 1 -
 configure.ac | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 7023988..f9de08c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,6 @@
 SUBDIRS = lib doc utils test-tool examples
 
 ACLOCAL_AMFLAGS =-I m4
-AUTOMAKE_OPTIONS = foreign subdir-objects
 AM_CPPFLAGS=-I. -I$(srcdir)/include "-D_U_=__attribute__((unused))" \
 	"-D_R_(A,B)=__attribute__((format(printf,A,B)))"
 AM_CFLAGS=$(WARN_CFLAGS)
diff --git a/configure.ac b/configure.ac
index e1d01c6..6d29af0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ AC_INIT(libiscsi, m4_esyscmd([grep 'Version:' ./packaging/RPM/libiscsi.spec.in 2
 AC_CONFIG_SRCDIR([lib/init.c])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
 AM_SILENT_RULES
 LT_INIT
 
-- 
2.0.0

